backend/middlewares/auth.ts

```diff
@@ -21,6 +30,81 @@ export const getAccessToken = (req: Request): string | null => {
   return null;
 };
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+const callResolver = async (
+  resolver: IMiddlewareFunction<any, ExpressContext>,
+  resolve: Resolve,
+  parent: any,
+  args: { [key: string]: any },
+  context: ExpressContext,
+  info: GraphQLResolveInfo,
+  // eslint-disable-next-line consistent-return
+) => {
+  if ("resolve" in resolver) {
+    return resolver.resolve?.(resolve, parent, args, context, info);
+  }
+  if (typeof resolver === "function") {
+    return resolver(resolve, parent, args, context, info);
+  }
+};
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+export const and = (
+  firstFn: IMiddlewareFunction<any, ExpressContext>,
+  secondFn: IMiddlewareFunction<any, ExpressContext>,
+) => {
+  return async (
+    resolve: Resolve,
+    parent: any,
+    args: { [key: string]: any },
+    context: ExpressContext,
+    info: GraphQLResolveInfo,
+  ) => {
+    const resolveFirst = async () => {
+      return callResolver(secondFn, resolve, parent, args, context, info);
+    };
+    return callResolver(firstFn, resolveFirst, parent, args, context, info);
+  };
+};
+
+/* Determine if the user whose role is to be changed does not match the currently 
+logged in user. 
+ * Note: userIdField is the name of the request parameter containing the requested userId */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+export const isAuthorizedToChangeRole = (userIdField: string) => {
+  return async (
+    resolve: (
+      parent: any,
+      args: { [key: string]: any },
+      context: ExpressContext,
+      info: GraphQLResolveInfo,
+    ) => any,
+    parent: any,
+    args: { [key: string]: any },
+    context: ExpressContext,
+    info: GraphQLResolveInfo,
+  ) => {
+    const accessToken = getAccessToken(context.req);
+    const authorized =
+      accessToken &&
+      (await authService.isAuthorizedToChangeRole(
+        accessToken,
+        args[userIdField],
+      ));
+
+    if (!authorized) {
+      throw new AuthenticationError(
+        "Failed authentication and/or authorization by userId",
```

Comment:
Could we change this error message to be something more descriptive? Something like "User ID cannot change its own role" or something.